Adapter.C

```diff
@@ -923,6 +924,60 @@ void preciceAdapter::Adapter::setupCheckpointing()
     ACCUMULATE_TIMER(timeInCheckpointingSetup_);
 }
 
+void preciceAdapter::Adapter::pruneCheckpointedFields()
+{
+    // Check if checkpointed fields exist in OpenFOAM registry
+    // If not, remove them from the checkpointed fields vector
+
+    word obj;
+    std::vector<word> regFields;
+    std::vector<word> toRemove;
+
+#undef doLocalCode
+#define doLocalCode(GeomField, GeomField_, GeomFieldCopies_)                                                           \
+    regFields.clear();                                                                                                 \
+    toRemove.clear();                                                                                                  \
+    /* Iterate through fields in OpenFOAM registry */                                                                  \
+    for (const word& obj : mesh_.sortedNames<GeomField>())                                                             \
+    {                                                                                                                  \
+        regFields.push_back(obj);                                                                                      \
+    }                                                                                                                  \
+    /* Iterate through checkpointed fields */                                                                          \
+    for (uint i = 0; i < GeomFieldCopies_.size(); i++)                                                                 \
+    {                                                                                                                  \
+        obj = GeomFieldCopies_.at(i)->name();                                                                          \
```

Comment:
What is the main reason we need the index here?
I guess we should be able to work with iterators, even if we only remove specific elements.
This already feels safer than before, but I think that the code can be further simplified and avoid the index calculations below.

Reply:
Good catch! I renamed the 'obj' variable to 'fieldObj' for clarity, and changed this loop to do without the index, because we do not actually need it here. Only place where the index is needed is in the last loop of this function.